haproxy/datadog_checks/haproxy/haproxy.py

```diff
@@ -146,6 +175,9 @@ def check(self, instance):
         if uptime is not None and uptime < startup_grace_period:
             return
 
+        if tables:
+            self._process_stick_table_metrics(tables, custom_tags=custom_tags)
```

Comment:
You don't pass the filters as parameters here.

Reply:
So there was supposedly a test for that, but it turns out `assert_all_metrics_covered` doesn't do what I expected, it only asserts that all the metric **names** that have been submitted were asserted, not that every metric submitted fell under an assertion.

This is fairly surprising I think, and not really documented as far as I can see